### ISSUE TITLE
in the Step 2: Individual (single-variant) analysis, Error in if (chr == 1) { : argument is of length zero

### ISSUE THREAD

**Issue by jingydz** [xihaoli/STAARpipeline-Tutorial#66]
It seems that I have properly executed the preparatory steps, as well as step zero and step one, and I have successfully obtained the files agds_dir.Rdata, Annotation_name_catalog.Rdata, and jobs_num.Rdata. Why would this error occur?

         used (Mb) gc trigger (Mb) max used (Mb)
Ncells 283209 15.2     664320 35.5   450929 24.1
Vcells 487102  3.8    8388608 64.0  1815713 13.9
Error in if (chr == 1) { : argument is of length zero
Execution halted

**Comment by xihaoli** [comment]
Hi @jingydz,

This error is indicating that your `chr` variable is missing (`NA`), which is likely due to the [following line](https://github.com/xihaoli/STAARpipeline-Tutorial/blob/main/STAARpipeline_Individual_Analysis.r#L39):

`arrayid <- as.numeric(commandArgs(TRUE)[1])`

Here, `arrayid` needs to be passed as an integer such that `chr` can be properly defined. In the tutorial, I am using it in conjunction with the [shell script](https://github.com/xihaoli/STAARpipeline-Tutorial/blob/main/batch%20jobs/STAARpipeline_Individual_Analysis.sh). Please feel free to adapt it in your case.

Thanks,
Xihao

**Comment by jingydz** [comment]
I now understand where my issue was. In the earlier steps, I modified the QC_label, changing the default 'annotation/filter' to 'annotation/info/QC_label', and in the subsequent step0 and step2, I ran them according to the default settings. After I changed to the new path, it is now running correctly, as in 1.0Association_Analysis_PreStep_change2info.R and 2STAARpipeline_Individual_Analysis_change2info.R where QC_label <- 'annotation/info/QC_label'. 

Thank you for your timely responses!
